### ISSUE TITLE
Fuzzy finder is not ignoring .gitignore files

### ISSUE THREAD

**Issue by coltonw** [atom/fuzzy-finder#89]
I have the "Exclude VCS Ignored Paths" setting on but fuzzy finder is NOT excluding them. My .gitignore looks like the following:
```
/webapp
/target
node_modules
shorthash.txt
/.idea/workspace.xml

```

Currently running latest Atom version which is using fuzzy finder 0.70.0

**Comment by izuzak** [comment]
Fuzzy finder is correctly excluding files for me. 

Can you share which steps _exactly_ reproduce the problem for you, starting from a `git clone` command which clones a repository containing the project and the .gitignore file? Also, please clarify which version of Atom you're using (`atom -v`) and which OS. (The latest version might be the latest released version, or the latest version built from `master` of atom/atom, and you didn't make that clear.)

**Comment by ismyrnow** [comment]
I'm experiencing the same on Windows 8.1, with version 0.187.0. I have confirmed that "Exclude VCS Ignored Paths" is checked in my settings. My gitignored files are greyed out in the tree view, but not ignored by fuzzy finder.

Steps to reproduce:

1. `git clone git@github.com:ismyrnow/Leaflet.groupedlayercontrol.git`
2. `npm install`
3. `atom --safe .`
4. `ctrl+t` then type `package.json`. You should only see one result.

**Comment by izuzak** [comment]
> You should only see one result.

Just followed the instructions and I indeed do see only one result -- the top-level package.json.

Can you clarify what you see, @ismyrnow?

![screen shot 2015-03-18 at 15 25 21](https://cloud.githubusercontent.com/assets/38924/6710740/0ac5af9e-cd83-11e4-99b2-5a9916128b14.png)

**Comment by ismyrnow** [comment]
Sorry. I see all of the package.json files in all subdirectories, including node_modules, which should be ignored.

![2015-03-18_1525](https://cloud.githubusercontent.com/assets/1127410/6717491/3f6ea93a-cd83-11e4-8dfb-ccd46412330c.png)

**Comment by izuzak** [comment]
Thanks for helping investigate this. Does this happen for any project and any ignore pattern, or just for this project and this pattern? Can you try changing the pattern in the .gitignore file to cover `bower.json` and see if that is handled correctly, for example?

**Comment by ismyrnow** [comment]
It happens for every project. I attempted a handful of other ignore patterns in my .gitignore file, including `node_modules`, `node_modules\`, `package.json`, and `bower.json` to name a few. When I save my .gitignore file and hit `ctrl+t`, it briefly says it is reindexing, then it shows all of the files as before.

It looks like it is ignoring all of the .gitignore patterns, regardless of how they are typed.

**Comment by coltonw** [comment]
I am having the same behavior as @ismyrnow on Windows 7.  My coworker is seeing the same issues on a Mac.

**Comment by Aedaeum** [comment]
I am also having the same issue. Windows 8.1. Atom 0.187.0. fuzzy-finder 0.70.0.

**Comment by darylng** [comment]
I have the same issue. Windows 8.1, Atom 0.188.0 fuzzy-finder 0.72.0 and `excludeVcsIgnoredPaths: true`

**Comment by darylng** [comment]
It appears to me that on my system in the code below, repo.getWorkingDirectory() returns path using `/` while this.rootPath uses `\`


```
if ((repo != null ? repo.getWorkingDirectory() : void 0) === this.rootPath) {
    this.repo = repo;
}
````

**Comment by nruth** [comment]
I'm also getting this on my Mac. 

```
atom -v
0.188.0
```

and fuzzy-finder is 0.72.0

Again, git ignores the files correctly and Atom is greying out the directories, but fuzzy finder is searching my tmp files and whatnot.

All my ignored file strings in atom are default or empty, I rely on the gitignore files doing everything. I have quite a lot of packages installed though, so should figure out how to test without them.

I've set up a minimal sample project where it happens:

```
git clone git@github.com:nruth/atom-fuzzy-finder-issue-demo.git 
cd atom-fuzzy-finder-issue-demo
mkdir tmp
touch tmp/shouldntseemee.txt
mkdir frontend/tmp
touch frontend/tmp/shouldntseeme.txt
git status
atom .
```
git status should say working directory clean.
Open in atom and try fuzzy finder, I get this:

![screen shot 2015-03-30 at 18 12 21](https://cloud.githubusercontent.com/assets/26158/6901957/9cc1291c-d708-11e4-9f7b-9cb25f36fbee.png)

**Comment by filimon-danopoulos** [comment]
I am also experiencing this on my Windows 8.1 machine. 
I am running Atom 0.188.0 and fuzzy-finder 0.72.0.

tree-view is handling ignored files correctly. I will try it from my Linux machine later and see if the issue is present there too.

**Comment by mjhm** [comment]
Also experiencing the problem. mac osx, atom 0.189.0, fuzzy-finder 0.72.0

**Comment by izuzak** [comment]
@nruth I just tried reproducing this with your instructions in https://github.com/atom/fuzzy-finder/issues/89#issuecomment-87757742, but wasn't able to. I'm using Atom 0.193.0-bcb8ca4 on OSX 10.10.2, and here's what I'm seeing in Atom:

![gi](https://cloud.githubusercontent.com/assets/38924/7202206/716c2300-e510-11e4-91b3-c119258b726e.gif)

Just to make sure -- you do have the "Exclude VCS Ignored Paths" config option enabled?

**Comment by coltonw** [comment]
"Exclude VCS Ignored Paths" config option is enabled for me.  Now running version 0.192.0 of Atom on Windows 7.

**Comment by blimmer** [comment]
Ditto. I have Exclude VCS Ignored Paths set and I still get things in my `.gitignore`. It makes fuzzy-finding very slow. 0.194.0.

![](https://i.imgur.com/Uk29BPn.png)
![](https://i.imgur.com/6COOFbK.png)
![](https://i.imgur.com/f3sFMcD.png)

**Comment by Srirangan** [comment]
Same for me:
```
0.194.0
ubuntu 14.10 and 15.04
```

Tried removing ~/.atom, ~/.config/atom and reinstalled with only default plugins.

Still the same bug.

**Comment by izuzak** [comment]
I was able to reproduce this with @nruth's instructions on Windows 8.1: https://github.com/atom/fuzzy-finder/issues/89#issuecomment-87757742. Also, I think @darylng's comment about the difference in `\` and `/` in `getWorkingDirectory` and `rootPath` is relevant: https://github.com/atom/fuzzy-finder/issues/89#issuecomment-86592076. /cc @maxbrunsfeld about [these few lines](https://github.com/atom/fuzzy-finder/blob/458347c3bbb23804927b90f5d414b56b198c1630/lib/load-paths-handler.coffee#L16-L18) related to that check.

**Comment by blimmer** [comment]
I'm not positive that this is the case, but I have a theory on what causes this.

If I start atom **before** I launch my development server (which writes to a .gitignore-d directory), I don't see the problem. However, when I launch atom and then launch my development server, I notice the issue.

Is it possible that atom will index any files that appear in the workspace after the initial load, even if they're gitignored?

**Comment by maxbrunsfeld** [comment]
@kevinsawicki or @zcbenz might be more familiar with this. @darylng has [reported](https://github.com/atom/fuzzy-finder/issues/89#issuecomment-86592076) that [`Repository::getWorkingDirectory`](https://github.com/atom/git-utils#repositorygetworkingdirectory) is returning a `/`-delimited path on windows.

**Comment by darylng** [comment]
Thanks everyone! Works for me now.

**Comment by Aedaeum** [comment]
I just want to confirm it works for me too :smile:

**Comment by coltonw** [comment]
Thanks everyone for their help! Works for me as well!
:tada:

Looks like it made it into Atom Version 0.196.0 for anyone wondering if this should be working for them or not.

**Comment by Pugio** [comment]
I'm getting this issue on 1.2.4 when I open up a subfolder of a git project. Using @nruth 's sample repo (and test files). If I do

```
cd frontend
atom .
```

I still see tmp/shouldntseeme.txt, in the tree-view, fuzzy-finder, and full project search. This is a problem, as I do a lot of development in a subfolder, but Atom doesn't seem to honor .gitignores not in the project root.

**Comment by lee-dohm** [comment]
@Pugio, it sounds like the issue you're talking about is https://github.com/atom/atom/issues/2203

**Comment by Pugio** [comment]
@lee-dohm : So it would seem, thanks. Guess I wasn't searching broadly enough.

**Comment by Wulf** [comment]
For any others having this issue: restarting atom worked for me.

**Comment by HeyHugo** [comment]
I think I've encountered a case where fuzzy finder still isn't respecting .gitignore:
If you have a project like this:
```
somerepo
  subfolder
    node_modules
    .gitignore
```
and .gitignore is
```
node_modules
```

..and then you just open subfolder in atom. node_modules will be indexed and show up in fuzzy search. I guess this is because it won't see this as a git repo since .git will be 1 level up, but maybe it still should?

**Comment by lee-dohm** [comment]
@HeyHugo This is a well-documented exception to Atom's Git support. If you open `subfolder` Atom doesn't walk up the tree to see if that folder is a subfolder of a Git repository. If you want Atom to respect all of the Git rules, then you need to open the root of the repository in Atom.

**Comment by HeyHugo** [comment]
Ok I understand, thanks for the info.

**Comment by bbugh** [comment]
I am having this issue. Our `tmp` folder is ignored in `.gitignore`, but the `tmp` folder is still indexed. I am not loading Atom in a subfolder or anything, just loading the base Rails folder than includes `.git`. 

```
$ git status --ignored
# ... snip ...
Ignored files:
	tmp/
```

![image](https://user-images.githubusercontent.com/438465/35131977-a43b0a56-fc8e-11e7-9940-3ffaed57103f.png)

Atom 1.23.3
fuzzy-finder 1.7.3

**Comment by bbugh** [comment]
I figured out why this is happening. `fuzzy-finder` is not minding the `.gitignore` if the project folder is a git worktree. This is also happening for `tree-view` which is not a surprise. 

Here's the main `master` project folder: 

![image](https://user-images.githubusercontent.com/438465/35251117-56502736-ff9f-11e7-858e-f85d92fecec9.png)

The exact same file structure, but with a `master` as a worktree:

![image](https://user-images.githubusercontent.com/438465/35251113-50844a8a-ff9f-11e7-8511-39121968cba0.png)

This seems to be related to https://github.com/atom/atom/issues/8168 and https://github.com/atom/git-utils/pull/77

**Comment by bernatfortet** [comment]
I followed this article's recommendation: https://medium.com/@kandros/prevent-fuzzy-search-in-node-modules-folder-in-atom-editor-a1fb337c2f97

TL;DR go to `packages > fuzzy finder > settings > ignored names` and add `node_modules` 👌

**Comment by bbugh** [comment]
This is great! Atom is actually usable again. Thanks for the protip.
